packages/legend-extension-dsl-data-space/src/stores/query/DataSpaceTemplateQueryViewerStore.ts

```diff
@@ -0,0 +1,215 @@
+      this.groupId,
+      this.artifactId,
+      this.versionId,
+      createViewProjectHandler(this.applicationStore),
+      createViewSDLCProjectHandler(
+        this.applicationStore,
+        this.depotServerClient,
+      ),
+    );
+    const sourceInfo = {
+      groupId: projectInfo.groupId,
+      artifactId: projectInfo.artifactId,
+      versionId: projectInfo.versionId,
+      dataSpace: dataSpace.path,
+    };
+    const queryBuilderState = new DataSpaceQueryBuilderState(
+      this.applicationStore,
+      this.graphManagerState,
+      this.depotServerClient,
+      dataSpace,
+      executionContext,
+      (dataSpaceInfo: DataSpaceInfo) => {
+        if (dataSpaceInfo.defaultExecutionContext) {
+          this.applicationStore.navigationService.navigator.goToLocation(
+            generateDataSpaceTemplateQueryViewerRoute(
+              guaranteeNonNullable(dataSpaceInfo.groupId),
+              guaranteeNonNullable(dataSpaceInfo.artifactId),
+              LATEST_VERSION_ALIAS, //always default to latest
+              dataSpaceInfo.path,
+              this.templateQueryTitle,
+            ),
+          );
+        } else {
+          this.applicationStore.notificationService.notifyWarning(
+            `Can't switch data space: default execution context not specified`,
+          );
+        }
+      },
+      true,
+      dataSpaceAnalysisResult,
+      (ec: DataSpaceExecutionContext) => {
+        this.applicationStore.navigationService.navigator.updateCurrentLocation(
+          generateDataSpaceTemplateQueryViewerRoute(
+            this.groupId,
+            this.artifactId,
+            this.versionId,
+            dataSpace.path,
+            this.templateQueryTitle,
+          ),
+        );
+      },
+      undefined,
+      undefined,
+      projectInfo,
+      this.applicationStore.config.options.queryBuilderConfig,
+      sourceInfo,
+    );
+    queryBuilderState.setExecutionContext(executionContext);
+    queryBuilderState.propagateExecutionContextChange(executionContext);
+    queryBuilderState.initializeWithQuery(dataSpaceExecutableTemplate.query);
+    return queryBuilderState;
+  }
+
+  getPersistConfiguration(
+    lambda: RawLambda,
+    options?: { update?: boolean | undefined },
+  ): QueryPersistConfiguration {
+    return {
+      defaultName: options?.update
+        ? `${extractElementNameFromPath(this.dataSpacePath)}`
+        : `New Query for ${extractElementNameFromPath(this.dataSpacePath)}[${
+            this.templateQueryTitle
+          }]`,
+      decorator: (query: Query): void => {
+        query.id = uuid();
+        query.groupId = this.groupId;
+        query.artifactId = this.artifactId;
+        query.versionId = this.versionId;
+        query.taggedValues = [
+          createQueryDataSpaceTaggedValue(this.dataSpacePath),
+          createQueryClassTaggedValue(
+            guaranteeNonNullable(this.queryBuilderState?.class?.path),
```

Comment:
lets add a null check here for guaranteeNonNullable. 
and add the class tagged value only if the class is defined (supported query). 
Could we check this in other places where we do guaranteNonNullable on the class path when decorating the query